packages/generic-utils/src/create-deferred/create-deferred.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Return a promise that can be externally resolved
+ */
+export const createDeferred = <T>() => {
```

Comment:
I look forward to the day https://github.com/tc39/proposal-promise-with-resolvers graduates and we can rely on it!